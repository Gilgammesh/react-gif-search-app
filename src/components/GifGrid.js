import React from "react";
import PropTypes from "prop-types";
import GifGridItem from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

const GifGrid = ({ categoria }) => {
  const { data, loading } = useFetchGifs(categoria);

  const imagesList = data.map((ele) => <GifGridItem key={ele.id} {...ele} />);

  return (
    <>
      <hr />
      <h3 className="animate__animated animate__fadeIn">
        {categoria.toUpperCase()}
      </h3>
      {loading && (
        <h4 className="animate__animated animate__flush">Cargando ...</h4>
      )}
      <div className="card-grid">{imagesList}</div>
    </>
  );
};

GifGrid.propTypes = {
  categoria: PropTypes.string.isRequired,
};

export default GifGrid;
