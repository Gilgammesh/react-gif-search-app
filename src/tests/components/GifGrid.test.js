import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas unitarias en el componente GifGrid.js", () => {
  const categoria = "naruto";
  const imgs = [
    { id: 1, title: "titulo1", url: "https://www.dominio.com/titulo1.gif" },
    { id: 2, title: "titulo2", url: "https://www.dominio.com/titulo2.gif" },
    { id: 3, title: "titulo3", url: "https://www.dominio.com/titulo3.gif" },
    { id: 4, title: "titulo4", url: "https://www.dominio.com/titulo4.gif" },
  ];
  /************************************************************************************/
  test("se debe mostrar el componente <GifGrid /> correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid categoria={categoria} />);
    expect(wrapper).toMatchSnapshot();
  });
  /************************************************************************************/
  test("se debe mostrar items cuando se cargan imagenes con useFetchGifs", () => {
    useFetchGifs.mockReturnValue({
      data: imgs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid categoria={categoria} />);
    const h4 = wrapper.find("h4");
    expect(h4.exists()).toBe(false);
    const comp = wrapper.find("GifGridItem");
    expect(comp.length).toBe(imgs.length);
  });
});
