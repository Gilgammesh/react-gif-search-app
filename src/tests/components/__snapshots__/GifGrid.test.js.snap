// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente GifGrid.js se debe mostrar el componente <GifGrid /> correctamente 1`] = `
<Fragment>
  <hr />
  <h3
    className="animate__animated animate__fadeIn"
  >
    NARUTO
  </h3>
  <h4
    className="animate__animated animate__flush"
  >
    Cargando ...
  </h4>
  <div
    className="card-grid"
  />
</Fragment>
`;
